diff --git a/.gitignore b/.gitignore
index d0d0163..bd7a370 100644
--- a/.gitignore
+++ b/.gitignore
@@ -2,12 +2,13 @@
 # Created by https://www.toptal.com/developers/gitignore/api/windows,visualstudiocode,python,blender,executable
 # Edit at https://www.toptal.com/developers/gitignore?templates=windows,visualstudiocode,python,blender,executable
 
+tests/
+
 ### Blender ###
 # Ignore temporary Blender files.
 *.blend[0-9]*
 
 ### Executable ###
-/.releases
 *.app
 *.bat
 *.cgi
diff --git a/audio/ambient.mp3 b/audio/ambient.mp3
deleted file mode 100644
index 7a79629..0000000
Binary files a/audio/ambient.mp3 and /dev/null differ
diff --git a/enemies.py b/enemies.py
index 3aaa94a..36cfc8b 100644
--- a/enemies.py
+++ b/enemies.py
@@ -31,6 +31,24 @@ class Enemy(Entity):
             "texture_scale",
             self.texture_scale
         )
+        
+        char = self.model.find("**/+Character").node()
+        part_bundle = char.getBundle(0)
+
+        self.anim_controls = {}
+
+        for node in self.model.findAllMatches("**/+AnimBundleNode"):
+            bundle = node.node().getBundle()
+
+            control = part_bundle.bindAnim(
+                bundle,
+                part_bundle.HMF_ok_anim_extra |
+                part_bundle.HMF_ok_part_extra |
+                part_bundle.HMF_ok_wrong_root_name,
+                PartSubset(),
+            )
+
+            self.anim_controls[bundle.getName()] = control
 
     def update_entity(self, dt):
         pass
@@ -70,7 +88,7 @@ class Staticon(Enemy):
         self.speed=speed
         self.ai_active = ai_active
         self.base_color = base_color
-        self.texture = generate_noise_texture("hidden_entity_"+str(random.randint(0,1000000)))
+        self.texture = generate_noise_texture("staticon_"+str(random.randint(0,1000000)))
 
         self.set_shader_input(
             "texture_scale",
@@ -85,24 +103,6 @@ class Staticon(Enemy):
         self.attack_range = attack_range
         self.awareness_range = awareness_range
 
-        char = self.model.find("**/+Character").node()
-        part_bundle = char.getBundle(0)
-
-        self.anim_controls = {}
-
-        for node in self.model.findAllMatches("**/+AnimBundleNode"):
-            bundle = node.node().getBundle()
-
-            control = part_bundle.bindAnim(
-                bundle,
-                part_bundle.HMF_ok_anim_extra |
-                part_bundle.HMF_ok_part_extra |
-                part_bundle.HMF_ok_wrong_root_name,
-                PartSubset(),
-            )
-
-            self.anim_controls[bundle.getName()] = control
-    
         self.anim_controls["attack"].setPlayRate(0.75)
         self.anim_controls["walking"].setPlayRate(2)
         self.anim_controls["idle"].play()
@@ -164,7 +164,7 @@ class Staticon(Enemy):
                 lerp(self.health/self.max_health, self.base_color.v, 1-self.damage_flash),
             )
             
-            self.texture = generate_noise_texture("hidden_entity_"+str(random.randint(0,1000000)))
+            self.texture = generate_noise_texture("staticon_"+str(random.randint(0,1000000)))
         elif self.health != 0:
             self.damage_flash = 0
             self.color = self.base_color
@@ -225,32 +225,13 @@ class Obeliskus(Enemy):
             max_health=max_health,
             enabled=enabled
         )
-        
-        char = self.model.find("**/+Character").node()
-        part_bundle = char.getBundle(0)
-
-        self.anim_controls = {}
-
-        for node in self.model.findAllMatches("**/+AnimBundleNode"):
-            bundle = node.node().getBundle()
-
-            control = part_bundle.bindAnim(
-                bundle,
-                part_bundle.HMF_ok_anim_extra |
-                part_bundle.HMF_ok_part_extra |
-                part_bundle.HMF_ok_wrong_root_name,
-                PartSubset(),
-            )
-
-            self.anim_controls[bundle.getName()] = control
 
         self.ai_active = ai_active
         self.speed=speed
         self.attack_range = attack_range
         self.awareness_range = awareness_range
         self.attacked_timer = 0
-        self.anim_controls["hit"].setPlayRate(0.5)
-        self.anim_controls["idle"].play()
+        self.attacking_timer = 0
         
         if self.enabled:
             print("a obeliskus has been summoned:")
@@ -296,64 +277,54 @@ class Obeliskus(Enemy):
             )
             dist = max(to_self.length() - self.scale.length() / 2, 0)
             if self.awareness_range > dist:
-                direction = to_self.normalized()
                 if self.attacked_timer <= 0:
                     target_rotation_y = math.degrees(math.atan2(to_self.x, to_self.z))
                     angle_diff = abs(((target_rotation_y % 360) - (self.rotation_y % 360) + 180) % 360 - 180)
                     if angle_diff > 4:
-                        print(angle_diff)
                         self.rotation_y = lerp_angle(self.rotation_y, target_rotation_y, time.dt * 5)
                     else:
                         if dist <= self.attack_range*self.scale.length():
-                            if not self.anim_controls["bite"].is_playing():
+                            if self.attacking_timer <= 0:
                                 self.anim_controls["bite"].play()
+                                self.attacking_timer = 2.5
+                            else:
+                                self.attacking_timer -= dt
                         else:
-                            self.position -= direction * min(
+                            forward = Vec3(
+                                math.cos(-math.radians(self.rotation_y)),
+                                0,
+                                math.sin(-math.radians(self.rotation_y))
+                            )
+                            
+                            self.position += forward * min(
                                 self.speed * dt * self.scale.length() * (self.health/self.max_health),
                                 dist
                             )
-
-                            if not self.anim_controls["move"].is_playing():
-                                self.anim_controls["move"].play()
-
-            if self.attack_range*self.scale.length() < dist:
-                if self.anim_controls["bite"].get_frame() < 15:
-                    self.anim_controls["bite"].stop()
-                
-                elif 15 < self.anim_controls["bite"].get_frame() < 27:
-                    self.anim_controls["bite"].stop()
-
-                elif 27 < self.anim_controls["bite"].get_frame() < 41:
-                    self.anim_controls["bite"].stop()
-
-                if self.anim_controls["bite"].get_frame() in [15, 27, 41]:
-                    self.attack(5)
-
             dx = player.x - self.x
             dz = player.z - self.z
 
             target_rotation_y = math.degrees(math.atan2(dx, dz))
             angle_diff = ((target_rotation_y % 360) - (self.rotation_y % 360) + 180) % 360 - 180
 
-            if (dist <= 1 or self.attacked_timer > 0) and not self.anim_controls["bite"].is_playing() or angle_diff > 4:
-                self.anim_controls["idle"].play()
+    def update_texture(self):
+        self.texture = generate_triangle_texture("obeliskus_"+str(int(self.position.x))+"_"+str(int(self.position.z)))
 
     def damage(self, damage):
         if super().damage(damage) != "":
             self.attacked_timer = 6 + self.attacked_timer/3
-            self.anim_controls["hit"].setPlayRate(0.5*self.health/self.max_health)
 
 class Maime(Enemy):
     def __init__(self, item: Item, speed: float = 4, size: int = 1, attack_range: int = 1, awareness_range: int = 10, max_health: int = 100, ai_active: bool = True, exposed: bool = False, enabled: bool = False):
         self.texture_scale = Vec2(15/item.scale.x, 15/item.scale.y)
         super().__init__(
-            item.model,
+            model="models/maime.glb",
             position=item.position,
             scale=item.scale,
             max_health=max_health,
             enabled=enabled
         )
         
+        self.model = item.model
         self.health = max_health
         self.awareness_range = awareness_range
         self.size = size
@@ -376,11 +347,11 @@ class Maime(Enemy):
         if self.enabled:
             print("a maime has been summoned:")
             print("    item disguised:", self.model)
-            print("        MODEL:", self.itemData[0])
+            print("        MODEL: item/", item.label.text.lower().replace(" ", "_"))
             print("        SCALE:", self.itemData[1])
-            # print("    available animation:")
-            # print("\n        ".join(self.anim_controls.keys()).upper())
-            # print("    animation:", "idle")
+            print("    available animation:")
+            print("\n        ".join(self.anim_controls.keys()).upper())
+            print("    animation:", "idle")
             print("    stats:")
             print("        SPEED:", self.speed)
             print("        SIZE:", self.scale.length())
@@ -423,18 +394,31 @@ class Maime(Enemy):
                     0,
                     self.z - player.z
                 )
+
                 dist = max(to_self.length() - self.scale.length() / 2, 0)
-                print(dist)
                 direction = to_self.normalized()
                 if dist > 0.05:
                     if dist <= self.attack_range*self.scale.length():
-                        pass
+                        if not self.anim_controls["attack"].is_playing():
+                            self.anim_controls["attack"].play()
                     else:
                         self.position -= direction * min(
                             self.speed * dt * self.scale.length(),
                             dist
                         )
 
+                        if not self.anim_controls["walk"].is_playing():
+                            self.anim_controls["walk"].play()
+                            
+            if self.anim_controls["attack"].get_frame() < 15:
+                self.anim_controls["attack"].stop()
+
+            if self.anim_controls["attack"].get_frame() == 15:
+                self.attack(5)
+
+            if dist <= 1 and not self.anim_controls["attack"].is_playing():
+                self.anim_controls["idle"].play()
+
 def trigger_screen_shift():
     global screen_shift_strength
     screen_shift_strength = 0.5
diff --git a/entities.py b/entities.py
index 5716209..62a7817 100644
--- a/entities.py
+++ b/entities.py
@@ -7,7 +7,8 @@ class Player(Entity):
             rotation=(0,0,0),
             y=0.3,
             unlit=True,
-            shader=triplanar_shader
+            shader=triplanar_shader,
+            enabled=False
         )
     
         self.speed = speed
@@ -112,7 +113,7 @@ class Player(Entity):
                     self.animation = "backwards"
 
                 if self.prev_frame_num != self.anim_controls[self.animation].getFrame() and self.anim_controls[self.animation].getFrame() % (self.anim_controls[self.animation].getNumFrames()//2) == 5:
-                    Audio(Path("audio/step.mp3"))
+                    Audio("audio/step.mp3", 2)
 
                 self.prev_frame_num = self.anim_controls[self.animation].getFrame()
                 self.position += movement * (self.speed * (2 if held_keys["left shift"] else 1)) * dt
@@ -154,7 +155,7 @@ class Player(Entity):
         self.pump_stroke_dist = 0.0
         self.pump_strokes = 0
         self.shotgun_ammo_count -= 1
-        Audio(Path("audio/shotgun/clink.mp3"))
+        Audio("audio/shotgun/clink.mp3")
         
         # ----- bullets hit detection with spread ----------------------------
         bullets_hit: dict[Entity, float] = {}
@@ -381,12 +382,10 @@ class ShowDialog(Entity):
         dt = time.dt
         if self.dialog_timer < len(self.dialog_text) + 2/self.dialog_speed:
             self.dialog_timer += dt/self.dialog_speed
-            # if int(self.dialog_timer) == int(self.dialog_timer - dt/self.dialog_speed):
-            #     Audio("audio/dialog_pop.mp3", autoplay=True, auto_destroy=True)
-            self.label.text = self.dialog_text[:int(min(self.dialog_timer, len(self.dialog_text)))]
-            player.can_move = (self.dialog_timer >= len(self.dialog_text))
+            if self.label.text != self.dialog_text[:int(min(self.dialog_timer, len(self.dialog_text)))]:
+                Audio("audio/dialog_pop.mp3", autoplay=True, auto_destroy=True)
+                self.label.text = self.dialog_text[:int(min(self.dialog_timer, len(self.dialog_text)))]
         else:
-            player.can_move = True
             self.dialog_callback()
             
             destroy(self.label)
@@ -401,8 +400,6 @@ class ShowDialog(Entity):
     def dialog_callback(self):
         pass
 
-player.can_move = False
-
 tutorial = Text(
     text="",
     color=color.yellow,
@@ -507,3 +504,51 @@ win_text = Text(
     position=(-0.2,0),
     scale=1
 )
+
+ground = Entity(
+    model='cube',
+    scale=(width, 0, height),
+    shader=triplanar_shader,
+    position=(center_x, center_y),
+    texture=generate_noise_texture("world"),
+    color=color.rgba(0.5,0.5,0.5,0),
+    unlit=True
+)
+
+
+void = Entity(
+    model='plane',
+    texture=generate_noise_texture("void"),
+    scale=200,
+    y=-0.1,
+    texture_scale=(5, 5),
+    color=color.rgb(1, 1, 1, 0),
+    unlit=True
+)
+
+void_fade_in = void.fade_in(1/16, duration=4)
+
+void_noise_timer = 0
+
+
+ground.set_shader_input(
+    "position", Vec2(0)
+)
+
+ground.set_shader_input(
+    "texture_scale", Vec2(0.05)
+)
+
+title_text = Text(
+    text="ONE LUCID NIGHT",
+    origin=(0, -0.35),
+    position=(0, 0.35),
+    scale=5,
+    color=color.red
+)
+
+start_game_text = Text(
+    text="press space to start",
+    origin=(0, -0.45),
+    position=(0, -0.45)
+)
\ No newline at end of file
diff --git a/imports.py b/imports.py
index 6e56aaf..69edc16 100644
--- a/imports.py
+++ b/imports.py
@@ -1,9 +1,9 @@
 import math
 import random
 import json
-from PIL import Image
+from PIL import Image, ImageDraw
 import numpy as np
 import hashlib
 from ursina import *
-
-from panda3d.core import PartSubset
\ No newline at end of file
+from panda3d.core import PartSubset
+from scipy.spatial import Delaunay
\ No newline at end of file
diff --git a/main.py b/main.py
index 6b4b8b7..c1badcd 100644
--- a/main.py
+++ b/main.py
@@ -20,39 +20,15 @@ from runes import *
 # World
 # -----------------------------
 
-ground = Entity(
-    model='cube',
-    scale=(width, 0, height),
-    shader=triplanar_shader,
-    position=(center_x, center_y),
-    texture=generate_noise_texture("world"),
-    color=color.rgba(0.5,0.5,0.5,0),
-    unlit=True,
-)
-
-
-void = Entity(
-    model='plane',
-    texture=generate_noise_texture("void"),
-    scale=200,
-    y=-0.1,
-    texture_scale=(5, 5),
-    color=color.rgb32(16, 16, 16),
-    unlit=True
-)
-
-void_noise_timer = 0
-
-ground.set_shader_input(
-    "position", Vec2(0)
-)
-
-ground.set_shader_input(
-    "texture_scale", Vec2(0.05)
-)
-
 def input(key):
-    global enemies, tutorial_ended
+    global enemies, tutorial_ended, main_menu
+    if main_menu:
+        if key == "space":
+            print("starting game")
+            main_menu = False
+            start_game()
+        return
+
     if not player.reloading and player.enabled:
         if key == 'left mouse down' and player.shotgun_ammo_count > 0:
             player.shoot()
@@ -151,16 +127,16 @@ if os.path.exists(wave_filepath):
     wave_filepath = wave_filepath[2:]
 
 
-def element_number_converter(num) -> list[float]:
-    if isinstance(num, str):
-        return {"l": left, "r": right, "t": top, "b": bottom}[num]
-    elif isinstance(num, dict):
-        minimum = element_number_converter(num["min"])
-        maximum = element_number_converter(num["max"])
+def element_number_converter(value) -> float:
+    if isinstance(value, str):
+        return {"l": left, "r": right, "t": top, "b": bottom}[value]
+    elif isinstance(value, dict):
+        minimum = element_number_converter(value["min"])
+        maximum = element_number_converter(value["max"])
         return random.uniform(minimum, maximum)
-    elif isinstance(num, list):
-        num = [element_number_converter(p) for p in num]
-        return random.choice(num)
+    elif isinstance(value, list):
+        return random.choice([element_number_converter(v) for v in value])
+    return value
 
 waves_data = json.load(open(wave_filepath))
 
@@ -169,16 +145,19 @@ for wave_num, wave_data in enumerate(waves_data):
     for element in wave_data:
         position = (element_number_converter(element["position"][0]), 1, element_number_converter(element["position"][1]))
         if element["type"] == "item":
-            name = random.choice(element["items"])
-            waves[-1]["items"].append(available_items[name](position=position))
+            elementIndex = random.choice(element["items"])
+            waves[-1]["items"].append({"items": available_items[elementIndex], "position": position})
         elif element["type"] == "rune":
-            name = random.choice(element["runes"])
-            waves[-1]["runes"].append(available_runes[name](position=position))
+            elementIndex = random.choice(element["runes"])
+            waves[-1]["runes"].append({"rune": available_runes[elementIndex], "position": position})
         elif element["type"] == "enemy":
-            name = random.choice(element["enemies"])
-            waves[-1]["enemies"].append(available_enemies[name](position=position))
-
-ground.fade_in(1, duration=3)
+            elementIndex = random.choice(element["enemies"])
+            
+            kwargs = {"enemy": available_enemies[elementIndex], "position": position}
+            if kwargs["enemy"] == Maime:
+                kwargs["item"] = {"entity": random.choice(available_items), "position": position}
+            
+            waves[-1]["enemies"].append(kwargs)
 
 def dialogCallback3():
     global items
@@ -187,6 +166,7 @@ def dialogCallback3():
     crosshair.color = color.rgb(crosshair.color.r, crosshair.color.g, crosshair.color.b, 1)
     tutorial.text = "TUTORIAL: GO TOUCH THE ITEM"
     items.append(ammoBox((random.uniform(1, 3), 0.3, random.uniform(0, 5)), True))
+    player.can_move = True
 
 def dialogCallback2():
     crosshair.enabled=True
@@ -210,12 +190,10 @@ def dialogCallback1():
     
     ShowDialog("and why do I have a shotgun?!").dialog_callback = dialogCallback2
 
-def update():
-    global screen_shift_strength, screen_fade_animation, void_noise_timer
+def update_game(dt: float):
+    global screen_shift_strength, screen_fade_animation
     global game_won, waiting_to_advance
 
-    dt = time.dt
-
     if screen_fade_animation > 0:
         screen_fade_animation -= dt/10
         screen_shift.color = color.rgba((power_lerp(1-screen_fade_animation) if screen_fade_animation > 0 else 0), 0, 0, 1 - screen_fade_animation)
@@ -258,11 +236,6 @@ def update():
         player.z
     )
 
-    void_noise_timer += dt
-    if void_noise_timer > 1.5:
-        void_noise_timer = 0
-        void.texture = generate_noise_texture("void_"+ str(int(random.random()*10)))
-
     alive_enemies = 0
     for entity in enemies:
         entity.update_entity(dt)
@@ -279,6 +252,26 @@ def update():
         runes.append(AdvancingRune(enabled=True))
         waiting_to_advance = True
 
-ShowDialog("where am I...").dialog_callback = dialogCallback1
+def start_game():
+    Audio("audio/ambient.wav", loop=True)
+    ground.fade_in(1, duration=3)
+    invoke(lambda: setattr(ShowDialog("where am I..."), "dialog_callback", dialogCallback1), delay=3)
+    void_fade_in.finish()
+    title_text.fade_out()
+    start_game_text.disable()
+    player.enable()
+    player.can_move = False
+
+def update():
+    global void_noise_timer
+    dt = time.dt
+
+    if not main_menu:
+        update_game(dt)
+
+    void_noise_timer += dt
+    if void_noise_timer > 1.5:
+        void_noise_timer = 0
+        void.texture = generate_noise_texture("void_"+ str(int(random.random()*10)))
 
 app.run()
\ No newline at end of file
diff --git a/models/obeliskus.glb b/models/obeliskus.glb
index 636ad9e..7e2595f 100644
Binary files a/models/obeliskus.glb and b/models/obeliskus.glb differ
diff --git a/models/obeliskus/animated.glb b/models/obeliskus/animated.glb
deleted file mode 100644
index bf7202c..0000000
Binary files a/models/obeliskus/animated.glb and /dev/null differ
diff --git a/models/obeliskus/entity.blend b/models/obeliskus/entity.blend
index 997a3b6..692b15b 100644
Binary files a/models/obeliskus/entity.blend and b/models/obeliskus/entity.blend differ
diff --git a/runes.py b/runes.py
index c8138a1..6291e4d 100644
--- a/runes.py
+++ b/runes.py
@@ -85,14 +85,16 @@ class AdvancingRune(Rune):
         
         wave_num += 1
         if wave_num-1 < len(waves):
-            for item in waves[wave_num-1]["items"]:
-                items.append(duplicate(item, enabled=True))
+            for itemData in waves[wave_num-1]["items"]:
+                items.append(duplicate(itemData.pop("item")(**itemData), enabled=True))
 
-            for enemy in waves[wave_num-1]["enemies"]:
-                enemies.append(duplicate(enemy, enabled=True))
+            for enemyData in waves[wave_num-1]["enemies"]:
+                if "item" in enemyData.keys():
+                    enemyData["item"] = enemyData["item"].pop("entity")(**enemyData["item"])
+                enemies.append(duplicate(enemyData.pop("enemy")(**enemyData), enabled=True))
 
-            for rune in waves[wave_num-1]["runes"]:
-                runes.append(duplicate(rune, enabled=True))
+            for runeData in waves[wave_num-1]["runes"]:
+                runes.append(duplicate(runeData.pop("runes")(**runeData), enabled=True))
         else:
             game_won = True
             screen_fade_animation = 1
diff --git a/util.py b/util.py
index 64b2f7b..80b5c84 100644
--- a/util.py
+++ b/util.py
@@ -15,6 +15,37 @@ def generate_noise_texture(seed: str, width=512, height=512):
     )
     return Texture(img)
 
+def generate_triangle_texture(seed: str, width=512, height=512):
+    data = seed.encode('utf-8')
+    seed = hashlib.sha512(data).digest().decode()
+    random = np.random.RandomState(seed)
+    
+    num_points = 150
+    points = random.rand(num_points, 2) * [width - 24, height - 24]
+    corners = np.array([[0, 0], [width, 0], [0, height], [width, height]])
+    all_points = np.vstack([points, corners])
+
+    tri = Delaunay(all_points)
+
+    image = Image.new("RGB", (width, height))
+    draw = ImageDraw.Draw(image)
+
+    for simplex in tri.simplices:
+        p1 = tuple(all_points[simplex[0]])
+        p2 = tuple(all_points[simplex[1]])
+        p3 = tuple(all_points[simplex[2]])
+
+        # Random RGB color
+        color = (
+            random.randint(0, 255),
+            random.randint(0, 255),
+            random.randint(0, 255),
+        )
+        draw.polygon([p1, p2, p3], fill=color)
+    
+    return Texture(draw)
+
+
 GROUND_NOISE_SCALE = 3.5
 
 BOUNDARY_REGION = (-25, -25, 25, 25)
@@ -95,4 +126,6 @@ game_over = game_won = tutorial_ended = False
 
 enemies: list = []
 items: list = []
-runes: list = []
\ No newline at end of file
+runes: list = []
+
+main_menu = True
\ No newline at end of file
diff --git a/waves.json b/waves.json
index 81bf22f..46c5131 100644
--- a/waves.json
+++ b/waves.json
@@ -1,9 +1,9 @@
 [
     [
         {
-            "type": "rune",
-            "runes": [0],
-            "position": [{"min": ""}]
+            "type": "enemy",
+            "enemies": [2],
+            "position": [[{"min": -10, "max": "l"}, {"min": 10, "max": "r"}], [{"min": 10, "max": "t"}, {"min": -10, "max": "b"}]]
         }
     ]
 ]
\ No newline at end of file
